### DISCUSSION TITLE
连接密码是什么都东西？

### DISCUSSION THREAD

**Discussion by 631068264** [antoor/antSword#328] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/8144089/51827458-6b72ef00-2324-11e9-84e3-4c35ce7f3d2d.png)
![image](https://user-images.githubusercontent.com/8144089/51827559-aecd5d80-2324-11e9-9276-4ca87c59acbb.png)

我是刚入门的新手，刚刚接触webshell

**OSX 10.14.2**
**php7.0**
```
<?php @eval($_POST['ant']) ?>
```
怎样改写`ant`关键字在antSword？

**Comment by Medicean** [comment, marked as answer]
所谓的 「连接密码」，其实是延续下来的一种叫法。因为这么叫的人多了，所以就都这么叫了。

那么「连接密码」的准确定义应该是什么呢？

首先要知道 「一句话 webshell」的原理是什么，一句话 webshell 之所以短小，但是功能却很强大，所依赖的正是 PHP, ASP, ASPX 这些脚本语言的「动态代码执行」这一特性，即 把字符串当作代码来执行。

拿你的这张图为例子：

![](https://user-images.githubusercontent.com/8144089/51827458-6b72ef00-2324-11e9-84e3-4c35ce7f3d2d.png)

`phpinfo();` 是字符串，我们把这个实际是PHP代码的字符串，发送到 webshell 服务端，再由 webshell 服务端执行并输出结果。

而这当中 **webshell 接收代码的入口参数的名称**，就是我们所说的「连接密码」了。

至于 `$_POST`, `$_GET`, `$_REQUEST` 这些到底是什么意思，我想你还需要看一下 PHP 的语法，然后再来继续，会更容易理解一些。

> 磨刀不误砍柴功, Good luck.

**Comment by 631068264** [comment]
谢谢指教 问题都解决了@Medicean
